### PR TITLE
[IMP] portal,*: update urls into nice urls

### PR DESCRIPTION
**Pull request by sauc-odoo**
*= hr_expense, im_livechat, mail, mrp, payment,
stock_picking_batch, test_mail

This commit modifies existing URLs to the new format 
so that the URL's become more human readable.

Task-3820230

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mail/controllers/mail.py [hunk 1]
```diff
@@ -219,17 +219,8 @@ def mail_thread_message_redirect(self, message_id, **kwargs):
             else:
                 raise Unauthorized()
 
-        url_params = {
-            'highlight_message_id': message_id,
-        }
         if message.model == 'discuss.channel':
-            url_params['active_id'] = message.res_id
-            url_params['action'] = 'mail.action_discuss'
+            url = f'/odoo/{message.res_id}/action-mail.action_discuss?highlight_message_id={message_id}'
```
Comment (jpp-odoo) [inline addons/mail/controllers/mail.py:223]: active_id in discuss is a little more special, they are alphanumeric, so they need to be in the queryString.

```suggestion
            url = f'/odoo/action-mail.action_discuss?active_id={message.res_id}&highlight_message_id={message_id}'
```

File: addons/im_livechat/views/im_livechat_chatbot_templates.xml [hunk 2]
```diff
@@ -17,7 +17,7 @@
                     <div groups="im_livechat.im_livechat_group_user" t-ignore="true"
                             class="alert alert-info alert-dismissible rounded-0 fade show d-print-none css_editable_mode_hidden mb-0">
                         <div t-ignore="true" class="text-center">
-                            <a t-attf-href="/web#view_type=form&amp;model=chatbot.script&amp;id=#{chatbot_data['id']}&amp;action=im_livechat.chatbot_script_action">
+                            <a t-attf-href="/odoo/action-im_livechat.chatbot_script_action/{chatbot_data['id']}">
```
Comment (jpp-odoo) [inline addons/im_livechat/views/im_livechat_chatbot_templates.xml:20]:
```suggestion
                            <a t-attf-href="/odoo/action-im_livechat.chatbot_script_action/#{chatbot_data['id']}">
```

**Review by jpp-odoo — CHANGES_REQUESTED** [review]

File: addons/im_livechat/views/im_livechat_chatbot_templates.xml (hunk 2 above)
Comment (sauc-odoo) [reply addons/im_livechat/views/im_livechat_chatbot_templates.xml]: Hello @jpp-odoo 
`<a t-attf-href="/odoo/action-im_livechat.chatbot_script_action/#{chatbot_data['id']}">`
Implied above changes, Please check it.
Thanks!!

**Comment by jpp-odoo** [comment]
@robodoo r+

**Pull request closed without merging**